src/corehost/cli/deps_resolver.cpp

```diff
@@ -126,22 +320,17 @@ void deps_resolver_t::get_dir_assemblies(
 //    Look for CoreCLR from the dependency list in the package cache and then
 //    the packages directory.
 //
-pal::string_t deps_resolver_t::resolve_coreclr_dir(
-    const pal::string_t& app_dir,
-    const pal::string_t& package_dir,
-    const pal::string_t& package_cache_dir)
+pal::string_t deps_resolver_t::resolve_coreclr_dir()
 {
     auto process_coreclr = [&]
         (bool is_portable, const pal::string_t& deps_dir, deps_json_t* deps) -> pal::string_t
     {
         pal::string_t candidate;
 
-        // Servicing override.
         if (deps->has_coreclr_entry())
         {
             const deps_entry_t& entry = deps->get_coreclr_entry();
-            trace::verbose(_X("Probing for CoreCLR package=[%s][%s] in servicing"), entry.library_name.c_str(), entry.library_version.c_str());
-            if (entry.is_serviceable && m_svc.find_redirection(entry.library_name, entry.library_version, entry.relative_path, &candidate))
+            if (probe_entry_in_configs(entry, &candidate))
```

Comment:
CoreCLR probe order, if I read correctly, looks wrong. We are going through the configs (which map to various probe paths) and then looking AppLocal and then FX instead of "dotnetextension;optimizationcache;fx;additionalprobepaths;applocal"